api/data_source.go

```diff
@@ -92,6 +92,10 @@ type DataSourceCreate struct {
 	Type     DataSourceType `jsonapi:"attr,type"`
 	Username string         `jsonapi:"attr,username"`
 	Password string         `jsonapi:"attr,password"`
+	// If true, syncs the schema after creating the data source. The client
+	// may set to false if the target data source's instance contains too many databases
+	// to avoid the request timeout.
+	SyncSchema bool `jsonapi:"attr,syncSchema"`
```

Comment:
Maybe we only need SyncSchema flag for Instances only. The instance is the one mapping to schemas. DataSources contain user/password only that doesn't change the way to get schema. For example, we don't need to sync schema while adding a read-only data source.

Reply:
Per-offline chat, the data source update might fix a previous broken data source and we need to sync schema afterwards.